interface/patient_file/encounter/forms.php

```diff
@@ -632,10 +638,22 @@ function divtoggle(spanid, divid) {
         echo "<div class='form_header_controls'>";
 
         // If the form is locked, it is no longer editable
-        if ( $esign->isLocked() ) {
-            echo "<a href=# class='css_button_small form-edit-button-locked' id='form-edit-button-".attr($formdir)."-".attr($iter['id'])."'><span>".xlt('Locked')."</span></a>";
+        if ($esign->isLocked()) {
+             $flag = false;
+             if($GLOBALS['enable_group_therapy']){
+                 if($acl_groups) $flag = true;
+             }
+             else{
+                 $flag = true;
+             }
+
+             if($flag){
+                 echo "<a href=# class='css_button_small form-edit-button-locked' id='form-edit-button-" . attr($formdir) . "-" . attr($iter['id']) . "'><span>" . xlt('Locked') . "</span></a>";
+             }
```

Comment:
$acl_groups = acl_check("groups","glog",false, 'write') ? true : false;
..
.
.
 if($GLOBALS['enable_group_therapy']){
                 if($acl_groups) $flag = true;
             }
             else{
                 $flag = true;
             }

This means that if a user have permission write  to groups log  the flag will become true. If not flag will be false. otherwise if there is no group acl the flag will be true and will continue with the code as it was. in the future we can use this segment of code to maybe give a global prmission to edit form. In this branch there only a group acl intent.